syntax = "proto3";

package data;

service DataService {
    rpc GetData (DataRequest) returns (DataResponse) {};
    rpc GetDataStream (DataRequest) returns (stream Record) {};
}

message DataRequest {
    string name = 1;
    int32 num_records = 2;
}

message DataResponse {
    repeated Record records = 1;
}

message Record {
    string string_val = 1;
    int32 int_val = 2;
    float float_val = 3;
}